### ISSUE TITLE
Domestic and Maritime Trade enabled when not the player's turn

### ISSUE THREAD

**Issue by lastmjs** [cs340tabyu/cs340Winter2017#708]
When logged in on a local machine with four clients, the fourth client is allowing the domestic and maritime modals can be seen, and they do not indicate that it isn't the player's turn.

Domestic trade should look like this when it is not the player's turn:

![screenshot from 2016-11-18 13 37 33](https://cloud.githubusercontent.com/assets/7000719/20445958/85eac798-ad94-11e6-9a90-7846e1877cd7.png)

But the client looks like this:

![screenshot from 2016-11-18 13 37 24](https://cloud.githubusercontent.com/assets/7000719/20445968/9053b6f4-ad94-11e6-8dcb-4933f6cd5916.png)

Maritime trade should look like this when it is not the player's turn:

![screenshot from 2016-11-18 13 38 24](https://cloud.githubusercontent.com/assets/7000719/20445976/a0fc4be2-ad94-11e6-897a-bb95fb0797bf.png)

But the client looks like this:

![screenshot from 2016-11-18 13 38 21](https://cloud.githubusercontent.com/assets/7000719/20445985/a70b690a-ad94-11e6-9b41-715f6e7c4272.png)

You can also send a domestic trade when it isn't your turn:

![screenshot from 2016-11-18 13 58 34](https://cloud.githubusercontent.com/assets/7000719/20446505/2915399c-ad97-11e6-92c6-c042a8ef8fbe.png)

**Comment by dansteren** [comment]
I also experienced this. I created a new game, created 4 players, went through the initial turns for everyone. Player 1 (Daniel) rolled. Then player 4 (Celeste) sent Player 1 (Daniel) a trade.
I declined this trade, then made a second and took a screenshot.
![player1](https://cloud.githubusercontent.com/assets/5455419/20452882/d18f80de-add1-11e6-83b8-9f591e9a6430.png)
![player4](https://cloud.githubusercontent.com/assets/5455419/20452884/d3535274-add1-11e6-8f4d-9571342273e6.png)
